### PR TITLE
Implemented a skeleton code for csr matrix class

### PR DESCRIPTION
**Pull request by hpleei**
1. Wrote a class skeleton for the CSR matrix
2. Defined methods/interfaces for the class: Constructor, print, getter
3. No destructor as it wouldn't need any dynamically allocated data structure for this object.
4. Using vector instead of plain array for more flexibilities and simplicity.

### CODE REVIEW

**Comment by bkochuna** [comment]
@hpleei this is going to have to wait because the implementation of the `SparseMatrix` class which this needs to extend is not yet completed (#47). I did this last night, and need to clean up the keeps before I push it out. This will happen today.

File: src/csr/csrsparsemat.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include <vector>
+#include <iostream>
+
+using namespace std;
+
+tempalte <typename T>
```
Comment (knbarton) [inline src/csr/csrsparsemat.hpp:6]:
```suggestion
template <typename T>
```

File: src/csr/csrsparsemat.hpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+class CrsSparseMatrix{
```
Comment (knbarton) [inline src/csr/csrsparsemat.hpp:7]:
```suggestion
class CsrSparseMatrix{
```

File: src/csr/csrsparsemat.hpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    private:
+        vector<vector<T>> crs_matrix;
```
Comment (knbarton) [inline src/csr/csrsparsemat.hpp:9]: Spelling
```suggestion
        vector<vector<T>> csr_matrix;
```

File: src/csr/csrsparsemat.hpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    public:
+        //Constructor - receives three inputs: element, col idx, row pointer vectors
+        //Builds CRS sparce matrix based on the input information
+        CrsSparseMatrix(const vector<T> & element_vec, const vector<int> & col_vec, const vector<int> & rowp_vec);
```
Comment (knbarton) [inline src/csr/csrsparsemat.hpp:13]:
```suggestion
        CsrSparseMatrix(const vector<T> & element_vec, const vector<int> & col_vec, const vector<int> & rowp_vec);
```

File: src/csr/csrsparsemat.hpp [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+        //Print the crs_matrix
+        void printMatrix();
+
+        //Return the crs_matrix
+        vector<vector<T>> get_crs_matrix();
```
Comment (knbarton) [inline src/csr/csrsparsemat.hpp:19]:
```suggestion
        vector<vector<T>> get_csr_matrix();
```

**Review by knbarton — CHANGES_REQUESTED** [review]
Thanks for getting this started! I'd like to look through the code more in depth later today, but for now I have some suggestions for spelling changes.

**Comment by knbarton** [comment]
I went ahead and made some changes to make the class so that it follows the testSparseMatrix class. I also added a csrTest to the test file to make sure the skeleton code works.

**Comment by knbarton** [comment]
This pull request now encompasses issue #19 as well, so I added the other assignee for that issue as a reviewer to look at the cmake files.

**Review by littlewatkins — COMMENTED** [review]
This all looks great to me! I don't have a good enough C++ background to speak to the specific code itself, but the content overall, like the header files and especially the test, is really nice. I give my approval, but I will wait for others thoughts.

**Review by bkochuna — APPROVED** [review]
If its good for @knbarton its good for me.

**Review by Qicang — APPROVED** [review]
It looks good to me to merge this branch.

**Pull request merged**